awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -603,7 +612,11 @@ def update_if_needed(self, existing_item, new_item, on_update=None, associations
 
         # If we change something and have an on_change call it
         if on_update is not None and self.json_output['changed']:
-            on_update(self, response['json'])
+            if response is None:
+                last_data = existing_item
+            else:
+                last_data = response['json']
+            on_update(self, last_data)
```

Comment:
@beeankha @john-westcott-iv I do want to avoid changing module_utils whenever possible

but I just rebased and this still accounts for a bug

```
===================================================================================== FAILURES ======================================================================================
___________________________________________________________________________ test_survey_spec_only_changed ___________________________________________________________________________
Traceback (most recent call last):
  File "/Users/alancoding/Documents/tower/awx_collection/test/awx/test_workflow_job_template.py", line 58, in test_survey_spec_only_changed
    }, admin_user)
  File "/Users/alancoding/Documents/tower/awx_collection/test/awx/conftest.py", line 146, in rf
    resource_module.main()
  File "/Users/alancoding/Documents/tower/awx_collection/plugins/modules/tower_workflow_job_template.py", line 234, in main
    on_create=on_change, on_update=on_change
  File "/Users/alancoding/Documents/tower/awx_collection/plugins/module_utils/tower_api.py", line 626, in create_or_update_if_needed
    return self.update_if_needed(existing_item, new_item, on_update=on_update, associations=associations)
  File "/Users/alancoding/Documents/tower/awx_collection/plugins/module_utils/tower_api.py", line 615, in update_if_needed
    on_update(self, response['json'])
TypeError: 'NoneType' object is not subscriptable
------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------

```

that's fixed with this change, may just be the first to hit it